harness/determined/core/_train.py

```diff
@@ -72,6 +72,7 @@ def report_training_metrics(
         steps_completed: int,
         metrics: Dict[str, Any],
         batch_metrics: Optional[List[Dict[str, Any]]] = None,
+        epoch: Optional[int] = None,
```

Comment:
any change to this will need @rb-determined-ai 's eyes, since it is core api

Reply:
iirc we had some sort of plan for this, and that is why it is `steps_completed` not `batches_completed`

Reply:
This is basically an additional x axis, and I'd much rather we start reporting it as a metric, rather than special-casing it and increasing the data model complexity.

Ultimately, W&B lets you select arbitrary X axes in their UI, and we also want to get there.  I think we should move in that direction.  I also think that adding `epoch` as a parameter here is effectively moving away from that direction.